plugins/deployment_manager/Spider.ts

```diff
@@ -1,211 +1,265 @@
 import { Contract } from 'ethers';
-import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { HardhatRuntimeEnvironment as HRE } from 'hardhat/types';
```

Comment:
The changes overall to spider are really hard to see in a diff. Need to review further, but overall, this returns more or less the same data as the previous version?

Reply:
Yeah I didn't really modify the interface (except adding the roots to return value). But it does basically rewrite the internal loop. There were a few issues before, and if you follow my WIP branches you'll see where I hit the main issue, when I was trying to preserve the way it was written and eventually couldn't and had to rewrite. I believe the killer was the fact that it was tracking visited by address, and not by alias, and there was a need to get a different interface for the same address under a different alias. The alias render stuff I had added before that to support the dynamic naming of `${symbol}:priceFeed`, and also the `Ctx` idea, and `artifact`. I added path after the rewrite (although attempted to add it earlier) to support the `${symbol}:admin` on FiatTokenProxy for crawling testnet aliases.

Reply:
This marks the third rewrite of Spider. A little confused how you track visited by alias, since we don't know the aliases until we actual spider the contracts, right?

Reply:
That's the crux of why I had to rewrite, we put the contract in by alias after we visit it, but we do have to do a little bit of work whenever we visit a node regardless to figure out how to proceed